tests/tools/combine-tools.test.js

```diff
@@ -0,0 +1,187 @@
+  const automatedToolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'automated-tools.json');
+
+  let manualTools;
+  let automatedTools;
+
+  let consoleErrorMock;
+
+  beforeAll(() => {
+    manualTools = readJSON(manualToolsPath);
+    automatedTools = readJSON(automatedToolsPath);
+
+    consoleErrorMock = jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterAll(() => {
+    if (fs.existsSync(toolsPath)) fs.unlinkSync(toolsPath);
+    if (fs.existsSync(tagsPath)) fs.unlinkSync(tagsPath);
+  });
+
+  it('should combine tools and create correct JSON files', async () => {
+    await combineTools(automatedTools, manualTools, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    expect(combinedTools).toHaveProperty('category1');
+
+    const tagsData = readJSON(tagsPath);
+    expect(tagsData).toHaveProperty('languages');
+    expect(tagsData).toHaveProperty('technologies');
+    expect(tagsData.languages).toContainEqual({
+      name: 'JavaScript',
+      color: 'bg-[#57f281]',
+      borderColor: 'border-[#37f069]'
+    });
+    expect(tagsData.languages).toContainEqual({
+      name: 'Python',
+      color: 'bg-[#3572A5]',
+      borderColor: 'border-[#3572A5]'
+    });
+    expect(tagsData.technologies).toContainEqual({
+      name: 'Node.js',
+      color: 'bg-[#61d0f2]',
+      borderColor: 'border-[#40ccf7]'
+    });
+    expect(tagsData.technologies).toContainEqual({
+      name: 'Flask',
+      color: 'bg-[#000000]',
+      borderColor: 'border-[#FFFFFF]'
+    });
+  });
+
+  it('should handle tools with missing language or technology', async () => {
+    const manualToolsWithMissingData = [
+      {
+        title: 'Tool C',
+        filters: {},
+        links: { repoUrl: 'https://github.com/asyncapi/tool-c' }
+      }
+    ];
+
+    await combineTools({}, manualToolsWithMissingData, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    expect(combinedTools).toHaveProperty('category1');
+  });
+
+  it('should sort tools alphabetically by title', async () => {
+    const manualToolsToSort = {
+      category1: {
+        description: 'Sample Category',
+        toolsList: [
+          {
+            title: 'Tool Z',
+            filters: { language: 'JavaScript' },
+            links: { repoUrl: 'https://github.com/asyncapi/tool-z' }
+          },
+          {
+            title: 'Tool A',
+            filters: { language: 'Python' },
+            links: { repoUrl: 'https://github.com/asyncapi/tool-a' }
+          }
+        ]
+      }
+    };
+
+    await combineTools(manualToolsToSort, {}, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    const toolTitles = combinedTools.category1.toolsList.map(tool => tool.title);
+    expect(toolTitles).toEqual(['Tool A', 'Tool Z']);
+  });
+
+
+  it('should log validation errors to console.error', async () => {
```

Comment:
Errors shouldn't be validated via console messages. You should formally return a error using `Promise.reject`.

Reply:
@vishvamsinh28 Have you worked on this?